### PR TITLE
Switch to a 'dev' and 'prod' build system in Grunt

### PR DESCRIPTION
**Pull request by aaronr**
This pull request adds two new modules (processhtml and focus) to allow us to more consistently have a dev and prod build.  Now all parts of the task chain have a dev and prod version... and grunt will default to dev.  
Also added a template in index.html so that a dev build will import the standard JS and a prod build will use the minified JS.
This will require a npm install to get the new modules.  
We need to figure out the gh-pages work quickly as this will break our ability to push the master branch directly to the gh-pages branch.  @alukach is working on integrating https://github.com/tschaub/grunt-gh-pages into our project (which can push the dist dir to the gh-pages branch).

### CODE REVIEW

**Comment by aaronr** [comment]
This is to address issue #45

**Comment by alukach** [comment]
So, looking at this odd thing where Travis built and deployed to `gh-pages`.

The way our system works is that when a build passes tests, Travis runs the `deploy-to-gh-pages.sh` script.  In the script, [these lines](https://github.com/cugos/drop-n-chop/blob/080ee563546c23a91662e643ac4ea28245e49e68/.travis/deploy-to-gh-pages.sh#L15-L18) are supposed to abort if you're not building/testing the `master` branch.  Those failed.

Re-reading the Travis docs, looks like I was confused by what the `TRAVIS_BRANCH` environment variable represented:

> TRAVIS_BRANCH:For builds not triggered by a pull request this is the name of the branch currently being built; whereas for builds triggered by a pull request this is the name of the branch **targeted by the pull request** (in many cases this will be master).

I assumed it meant that it was the branch being built, not to target pull request.  Will make a quick fix.

File: README.md [hunk 1]
```diff
@@ -37,6 +37,16 @@ All of this work is made possible by [CUGOS](http://cugos.org), an open-source g
 
 **NOTE:** Things are moving very quickly with DNC right now. We are still continuing to build out and refactor the architecture of the application. If you want to read more about our decision-making process take a look at some of our [meeting notes in the wiki](https://github.com/cugos/drop-n-chop/wiki/Meeting-Notes---04-11-2015).
 
+Run a local server with a development (not minified) version of the code:
+
```
Comment (alukach) [inline README.md]: Looks like we're missing the command here

Comment (alukach) [inline README.md]: Actually, maybe these additions to the `README.md` should be rolled into the text on line `36` of the file

**Comment by alukach** [comment]
:+1: :+1: :+1: :+1: :+1: :+1: :+1: 

Kind of a wild ride, that one...

**Pull request merged**